### PR TITLE
Handle missing credentials directory

### PR DESCRIPTION
**Pull request by eliran89c**

### CODE REVIEW

File: cmd/get.go [hunk 1]
```diff
@@ -42,6 +43,17 @@ func processCredentials(creds *aws.Credentials, app string) error {
 			return fmt.Errorf("expanding config file path: %v", err)
 		}
 
+		// Check if global.credentials-path home folder exist, if not,create it
+		CredsFileParentDir := filepath.Dir(path)
+		if _, err := os.Stat(CredsFileParentDir); os.IsNotExist(err) {
+			log.Printf(color.YellowString("%s does not exist, creating it now... "), CredsFileParentDir)
+
+			err = os.MkdirAll(CredsFileParentDir, 0751)
```
Comment (johananl) [inline cmd/get.go]: Out of curiosity, why `0751`?

File: cmd/get.go [hunk 2]
```diff
@@ -42,6 +43,17 @@ func processCredentials(creds *aws.Credentials, app string) error {
 			return fmt.Errorf("expanding config file path: %v", err)
 		}
 
+		// Check if global.credentials-path home folder exist, if not,create it
+		CredsFileParentDir := filepath.Dir(path)
+		if _, err := os.Stat(CredsFileParentDir); os.IsNotExist(err) {
+			log.Printf(color.YellowString("%s does not exist, creating it now... "), CredsFileParentDir)
```
Comment (johananl) [inline cmd/get.go]:
```suggestion
			log.Printf(color.YellowString("Credentials file %s does not exist - creating it"), CredsFileParentDir)
```

File: cmd/get.go [hunk 3]
```diff
@@ -42,6 +43,17 @@ func processCredentials(creds *aws.Credentials, app string) error {
...
+
+			err = os.MkdirAll(CredsFileParentDir, 0751)
+			if err != nil {
+				return fmt.Errorf("unable to create aws credentials folder: %v", err)
```
Comment (johananl) [inline cmd/get.go]:
```suggestion
				return fmt.Errorf("creating credentials directory: %v", err)
```

**Comment by johananl** [comment]
Oh, and if you could rename the PR and the commit to "Handle missing credentials directory" that would be great (`~/.aws` is the default, but the user can override this).

File: cmd/get.go (hunk 1 above)
Comment (eliran89c) [reply cmd/get.go]: Hi @johananl ,
changed to 755

**Comment by eliran89c** [comment]
Hi @johananl ,
thank you for your quick reply, I updated the code based on your suggestions.
regarding your question, you right `~/.aws` is been created automatically by aws-cli,
but the user can override the default directory
furthermore, it is possible that the user using Clisso without aws-cli

**Comment by johananl** [comment]
>If you don't add Go unit tests, at the very least I would verify the new functionality works on Linux, OSX and Windows (the file operations are the part which could have portability issues).

I've verified this on Linux and Windows. @eliran-allcloud have you verified this works on OSX? I can then merge this.

**Comment by eliran89c** [comment]
yes, verified on Linux/Mac and Windows(server) - working as expected

**Comment by johananl** [comment]
Great. Thanks for your contribution!

**Pull request merged**